### ISSUE TITLE
highlight.js in the nodejs version of TW5

### ISSUE THREAD

**Issue by welford** [Jermolene/TiddlyWiki5#1301]
This was already discussed here : https://groups.google.com/forum/#!topic/tiddlywikidev/m0QRODcm6mY

It is possible to get highlight.js to work under nodejs as it currently is. I have a working example forked from the original TW5. The differences are quite small and can be found here : 44ff25f05266c41f68c7924d12693d3216afe77c

The changes are in some places just getting the hljs modules to be loaded even when it isn't the browser version. The other changes are to highlightblock.js which, when running under node.js, passes through the source code to be highlighted to hljs.highlight instead of hljs.highlightBlock 

The class name "hljs" is also appended to the domNode in addition to the target language, as this is usually handled within hljs.highlightBlock (in highlight.pack.js)

I can't think of a good way of filtering out non hljs codeblocks, so currently i check for the presence of "/" which should filter out stuff like "text/plain" and then just do a silent fail on anything which makes it though the filter but is not a language registered with hljs.

**Comment by welford** [comment]
```
domNode.children[0].innerHTML = hljs.highlight(this.language, this.getAttribute("code")).value;
```

Should be 

```
domNode.children[0].innerHTML = hljs.fixMarkup(hljs.highlight(this.language, this.getAttribute("code")).value);
```

For tabs to render correctly.

**Comment by Jermolene** [comment]
Thanks @welford, a pull request would be great.

I think that the version of highlight.js that we're using has fallen quite a way behind the latest release. Would you be able to update us to the latest?

**Comment by welford** [comment]
I may have spoken a bit too soon. My original fix worked when exporting static pages using a batch file, though not when running a webpage from a server.

I have fix this issue in the pull request though #1347 by splitting the register language and hljs object setup into two stages. It now works when running the server/html/batch exporting. It introduces a handful of bytes worth of extra data to get this working, which I think is acceptable. And editing the original highlight.pack.js into two files is a simple enough task.

I have also updated the highlightjs code and the Tomorrow Night theme to the latest version.

**Comment by Jermolene** [comment]
Thanks @welford I'll respond on the pull request

**Comment by Jermolene** [comment]
Fixed in #1347
